ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java

```diff
@@ -834,19 +833,36 @@ public void testWritesToDisabledCompactionTable() throws Exception {
 
   private ShowCompactResponseElement generateElement(long id, String db, String table, String partition,
       CompactionType type, String state) {
-    return generateElement(id, db, table, partition, type, state, System.currentTimeMillis());
+    return generateElement(id, db, table, partition, type, state, false);
+  }
+
+  private ShowCompactResponseElement generateElement(long id, String db, String table, String partition,
+      CompactionType type, String state, boolean manuallyInitiatedCompaction) {
+    return generateElement(id, db, table, partition, type, state, manuallyInitiatedCompaction, System.currentTimeMillis());
   }
 
   private ShowCompactResponseElement generateElement(long id, String db, String table, String partition,
       CompactionType type, String state, long enqueueTime) {
+    return generateElement(id, db, table, partition, type, state, false, enqueueTime);
+  }
+
+  private ShowCompactResponseElement generateElement(long id, String db, String table, String partition,
+      CompactionType type, String state, boolean manuallyInitiatedCompaction, long enqueueTime) {
     ShowCompactResponseElement element = new ShowCompactResponseElement(db, table, type, state);
     element.setId(id);
     element.setPartitionname(partition);
     element.setEnqueueTime(enqueueTime);
 
-    String runtimeId = ServerUtils.hostname() + "-" + ThreadLocalRandom.current().nextInt();
```

Comment:
This was generating negative integers, that's why the number of initiator and worker hosts was 2 in testUpdateCompactionMetrics